qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDiagnostics.java

```diff
@@ -265,6 +277,24 @@ private void validateDataModel(Node parent, Template template, QuteValidationSet
 				case INCLUDE:
 					validateIncludeSection((IncludeSection) section, diagnostics);
 					break;
+				case CASE:
+				case IS:
+					Section parentSection = section.getParentSection();
+					if (!isMatchSwitchWhenSection(parentSection, section)) {
+						Range range = QutePositionUtility.createRange(parentSection.getStartTagNameOpenOffset(),
+								parentSection.getStartTagNameCloseOffset(), template);

```

Comment:
At first I suggest that you use QutePositionUtility.selectStartTag

After that I think it is better to set the error in #case or #if section start tag 
Instead of parent section. Indead if your template is just

{#case \"Apple\"}

You should have à NPE error because parentSection will be null.

Please use section instead of parentSection and pleqse write test with just 
{#case \"Apple\"}

Reply:
I reopen the conversation since it has been opened and not resolved.